tests/api/Jalasoft.TeamUp.Resumes.API.Tests/PostPostulationApiTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Jalasoft.TeamUp.Resumes.API.Tests
+{
+    using Jalasoft.TeamUp.Resumes.API.Controllers;
+    using Jalasoft.TeamUp.Resumes.Core.Interfaces;
+    using Jalasoft.TeamUp.Resumes.Models;
+    using Jalasoft.TeamUp.Resumes.ResumesException;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Moq;
+    using Xunit;
+
+    public class PostPostulationApiTests
+    {
+        private readonly Mock<IPostulationsService> mockPostulationsService;
+        private readonly DefaultHttpContext mockHttpContext;
+        private readonly PostPostulation postPostulation;
+
+        public PostPostulationApiTests()
+        {
+            this.mockPostulationsService = new Mock<IPostulationsService>();
+            this.mockHttpContext = new DefaultHttpContext();
+            this.postPostulation = new PostPostulation(this.mockPostulationsService.Object);
+        }
+
+        [Fact]
+        public void PostPostulation_ValidPostulation_Created()
+        {
+            var request = this.mockHttpContext.Request;
+            this.mockPostulationsService.Setup(service => service.PostPostulation(null)).Returns(new Postulation());
+            var response = this.postPostulation.Run(request);
+            var createdResult = Assert.IsType<CreatedResult>(response);
+            Assert.IsType<Postulation>(createdResult.Value);
+            Assert.Equal(201, createdResult.StatusCode);
+        }
+
+        [Fact]
+        public void PostPostulation_InvalidPostulation_BadRequest()
+        {
+            var request = this.mockHttpContext.Request;
+            this.mockPostulationsService.Setup(service => service.PostPostulation(null)).Throws(new ResumesException(ResumesErrors.BadRequest, new FluentValidation.ValidationException("BadRequest")));
```

Comment:
Your serviceMock do not know about badRequest (service layer do not know badRequest word), i guess you must throw fluentvalidation only. See https://github.com/fundev3/teamup-projects/blob/develop/tests/api/Jalasoft.TeamUp.Projects.API.Tests/PostInvitationTest.cs